actors/cloudbroker/cloudbroker__computenode/methodclass/cloudbroker_computenode.py

```diff
@@ -124,22 +122,30 @@ def enable(self, id, gid, message, **kwargs):
                                       errorcb=errorcb)
         return status
 
+    @auth(['level2', 'level3'], True)
+    def upgrade(self, id, message, force, **kwargs):
+        stack = self.models.Stack.get(id)
+        version = self.cb.rebootStack(stack, force)
+        stack.version = version
+        stack.save()
+        return "stack %s upgraded to version : %s" % (id, version)
+
     def _get_stack_machines(self, stackId, fields=None):
         machines = self.models.VMachine.objects(status__nin=['DESTROYED', 'ERROR'], stack=stackId)
         if fields:
             return machines.only(**fields)
         return machines
 
     @auth(['level2', 'level3'], True)
-    def maintenance(self, id, gid, vmaction, **kwargs):
+    def maintenance(self, id, vmaction, **kwargs):
         """
         :param id: stack Id
-        :param gid: Grid id
         :param vmaction: what to do with vms stop or move
         :return: bool
         """
-        if vmaction not in ('move', 'stop'):
-            raise exceptions.BadRequest("VMAction should either be move or stop")
+        title = ""
+        if vmaction not in ('move', 'stop', 'migrate'):
```

Comment:
Please only keep move or migrate and add a flag force from within the actions popup. Explain clearly in the warning message